modules/fedimint-lnv2-client/src/cli.rs

```diff
@@ -0,0 +1,77 @@
+use std::{ffi, iter};
+
+use clap::Parser;
+use fedimint_core::api::FederationApiExt;
+use fedimint_core::endpoint_constants::{
+    ADD_GATEWAY_ENDPOINT, GATEWAYS_ENDPOINT, REMOVE_GATEWAY_ENDPOINT,
+};
+use fedimint_core::module::ApiRequestErased;
+use fedimint_core::util::SafeUrl;
+use serde::Serialize;
+use serde_json::Value;
+
+use crate::LightningClientModule;
+
+#[derive(Parser, Serialize)]
+enum Opts {
+    /// Add a vetted gateway
+    Add { gateway: SafeUrl },
+    /// Remove a vetted gateway
+    Remove { gateway: SafeUrl },
+    /// List our own vetted gateways
+    List,
+}
+
+pub(crate) async fn handle_cli_command(
+    lightning: &LightningClientModule,
+    args: &[ffi::OsString],
+) -> anyhow::Result<serde_json::Value> {
+    let opts = Opts::parse_from(iter::once(&ffi::OsString::from("lnv2")).chain(args.iter()));
```

Comment:
Why is the `lnv2` prepended here?

Reply:
I copied this part from the meta module as @dpc suggested.

Reply:
I have a suspicion: `clap` might skip the first arg because it's the binary name when parsing command line args directly. Here we are only parsing a subset of the command line args, so need to add a dummy first one.

Reply:
@elsirion That's right.

We could consider passing from the higher level the name of the module itself to make it unnecessary, and maybe even allow modules to detect their own kind.